tests/kafkatest/tests/core/replication_test.py

```diff
@@ -128,7 +128,10 @@ def min_cluster_size(self):
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["controller"],
             security_protocol=["PLAINTEXT", "SASL_SSL"])
-    def test_replication_with_broker_failure(self, failure_mode, security_protocol, broker_type):
+    @matrix(failure_mode=["clean_shutdown", "hard_bounce"],
+            broker_type=["leader"],
+            security_protocol=["SASL_SSL"], client_sasl_mechanism=["PLAIN"], interbroker_sasl_mechanism=["PLAIN", "GSSAPI"])
```

Comment:
@ewencp I dont actually know the answer. I chose one sanity test with a single broker (consumer test) and the replication tests because they restart brokers making them useful for testing authentication. I chose two types (PLAIN, multi-mechanism) and two types of tests (`clean_bounce`, `hard_shutdown`), that is 4 tests in all. I could reduce to just `hard_shutdown` (2 tests) or just SASL/PLAIN with `hard_shutdown` (1 test). Or just do upgrade tests since they do restart as well. Suggestions welcome :-)